opm/output/OutputWriter.hpp

```diff
@@ -74,44 +75,28 @@ class OutputWriter {
      * This routine should be called before the first timestep (i.e. when
      * timer.currentStepNum () == 0)
      */
-    virtual void writeInit(const SimulatorTimerInterface &timer, const NNC& nnc = NNC()) = 0;
+    virtual void writeInit( const NNC& nnc = NNC() ) = 0;
 
     /*!
      * \brief Write a blackoil reservoir state to disk for later inspection with
      *        visualization tools like ResInsight
      *
-     * \param[in] timer          The timer providing time, time step, etc. information
-     * \param[in] reservoirState The thermodynamic state of the reservoir
-     * \param[in] wellState      The production/injection data for all wells
+     * \param[in] report_step           The current report step
+     * \param[in] current_posix_time    Seconds elapsed since epoch
+     * \param[in] seconds_elapsed       Seconds elapsed since simulation start
+     * \param[in] reservoirState        The thermodynamic state of the reservoir
+     * \param[in] wells                 Well data
      *
      * This routine should be called after the timestep has been advanced,
      * i.e. timer.currentStepNum () > 0.
      */
-    virtual void writeTimeStep(const SimulatorTimerInterface& timer,
-                               const SimulationDataContainer& reservoirState,
-                               const WellState& wellState,
-                               bool  isSubstep) = 0;
+    virtual void writeTimeStep( int report_step,
+                                time_t current_posix_time,
+                                double seconds_elapsed,
+                                data::Solution reservoirState,
+                                data::Wells,
+                                bool  isSubstep) = 0;
```

Comment:
We are semantically not really interested if this is a substep or not; the interesing question is:

1. Should I write summary data?
2. Should I write restart data?

The answer to the first question is "always" yes; although you can use the keyword `RPTONLY`[1] to essentially configure a `false` on this. For the second question we have logic like:

```C++
if (!isSubstep)
   printRestart();
``` 
But then I think it is better to name the variable something like `printRestart` - which is the real question; in exceptional cases that can also be set to true even for a substep - although I do not think our full stack supports that at the moment.

[1]. I do not think we support that - quite uncommon.